### PR TITLE
Fix `now dev` and other changes

### LINKED ISSUE #3: contents changed in original source code when running now dev
> **Note**: This was transferred from the official [now-builders](https://github.com/zeit/now-builders/issues/589) repo.

Not sure if I should post this here or in `now-cli` repo, but here it is:

When running `now dev` with `@now/rust` builder using `index.rs` entry point, it keeps changing the content of `Cargo.toml` and `index.rs`

#### Changes in `Cargo.toml`
```diff
[[bin]]
name = "index"
-path = "index.rs"
+path = "/Users/fatih/dev/paper/services/embed/github/index.rs"
```

#### Changes in `index.rs`
```diff
+use now_lambda::lambda;
+use std::error::Error;

// my handler code here

+fn main() -> Result<(), Box<dyn Error>> {
+    Ok(lambda!(handler))
+}
```

The problem is I'm not supposed to commit this changes, and I think this change should only be applied in private copy (somewhere in `/var/private`?).

Another problem is when running `now dev` again, it will add another template over and over again, it's possible that I get something like this

```diff
+use now_lambda::lambda;
+use std::error::Error;

+use now_lambda::lambda;
+use std::error::Error;

+use now_lambda::lambda;
+use std::error::Error;

// my handler code here

+fn main() -> Result<(), Box<dyn Error>> {
+    Ok(lambda!(handler))
+}

+fn main() -> Result<(), Box<dyn Error>> {
+    Ok(lambda!(handler))
+}


+fn main() -> Result<(), Box<dyn Error>> {
+    Ok(lambda!(handler))
+}
```

## My config

`now.json`
```json
{
  "builders": [
    {
      "src": "services/embed/**/index.rs",
      "use": "@now/rust@canary"
    },
  ]
}
```

## Version
```sh
❯ now --version
15.4.0
❯ uname -a
Darwin TVLK-FatihKalifa.local 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64
❯ cargo --version
cargo 1.35.0 (6f3e9c367 2019-04-04)
```

### PR DESCRIPTION
**Pull request by ekadas**
My goal with this PR was to fix `now dev` (fixes #3 and #9). While doing that I've also aimed to provide feature parity for how the official builders work (I.e: debug logs, support for utility libraries, support for parameterized routes). Each commit should have a further description and rationale.

I think one of the more controversial changes is fixing the version to 3 and removing the old Cargo.toml builld support (as per [this comment](https://github.com/mike-engel/now-rust/pull/14#discussion_r344658083)). Deploying with other versions did not work for my test repository, although I've seen that the unit tests still worked, so I'm not sure if removing completely is the right thing. My rationale for doing so is that it provides more of a similar experience to the official builders and a major version bump in `now-rust` should not cause issues for dependencies. I also saw `probes` in unit tests which is no longer documented nor does it work when running `now` from the fixture itself, so maybe unit tests have other behaviors to how the builder is meant to be consumed.

Thanks for your work on this so far, let me know if you have any questions or comments.

### CODE REVIEW

**Comment by jlricon** [comment]
I verified this works, thanks!

File: src/index.ts [hunk 1]
```diff
@@ -369,33 +280,4 @@ export async function prepareCache({
...
-		if (files[cargoTomlPath]) break;
-		const newPath = path.dirname(currentPath);
-		if (currentPath === newPath) break;
-		currentPath = newPath;
-	}
-
-	return cargoTomlPath;
-}
-
-export const getDefaultCache = ({ files, entrypoint }: BuildOptions) => {
```
Comment (mike-engel) [inline src/index.ts:391]: Did Zeit change their caching mechanism, or are we just giving up since it never seemed to work?

**Review by mike-engel — APPROVED** [review]
@ekadas Thank you so much for working on this, and thanks for your patience on this review! This looks really good, and I'm thinking this constitutes a 1.0.0 rc 😄

As for pinning to verion 3, I think that's ok as well, especially since this will definitely be a major bump.

I have just one question around caching, otherwise I'd like to merge this soon.

File: src/index.ts (hunk 1 above)
Comment (ekadas) [reply src/index.ts:391]: I belive they expect [`prepareCache`](https://github.com/zeit/now/blob/master/DEVELOPING_A_RUNTIME.md#preparecache) for the caching and do not expect `getDefaultCache`, so this seemed to be redundant, as `now-rust` also exposes `prepareCache`. I could have missed it somewhere, but a quick search on [`zeit/now`](https://github.com/zeit/now/search?q=getDefaultCache&unscoped_q=getDefaultCache) did not return results for `getDefaultCache`.

I think the caching could be improved though, as while locally caching seems to work, when deploying multiple functions the build seems to re-download and build cargo packages, so builds can take quite long. That does not really have a negative effect on the development experience and I don't think zeit's pricing model is based on build time, so I don't think it's anything urgent, just for the future.

**Comment by mike-engel** [comment]
@ekadas Thanks again for all your work on this. I changed my mind, and this is now available as `now-rust@2.0.0`!

**Pull request merged**